.github/workflows/checks.yaml

```diff
@@ -401,30 +346,12 @@ jobs:
           cc -o bzpipe bzpipe.c target/${{matrix.target}}/release/deps/libbz2_rs.so -I ../
           ./bzpipe < Cargo.toml | ./bzpipe -d > out.txt
           cmp -s Cargo.toml out.txt
-      - name: "cdylib: rust-allocator"
-        env:
-          LD_LIBRARY_PATH: "target/${{matrix.target}}/release/deps"
-        working-directory: libbzip2-rs-sys-cdylib
-        run: |
-          cargo build --release --target ${{matrix.target}} --no-default-features --features="rust-allocator"
-          cc -o bzpipe bzpipe.c target/${{matrix.target}}/release/deps/libbz2_rs.so -I ../
-          ./bzpipe < Cargo.toml | ./bzpipe -d > out.txt
-          cmp -s Cargo.toml out.txt
-      - name: "cdylib: no_std"
-        env:
-          LD_LIBRARY_PATH: "target/${{matrix.target}}/release/deps"
-        working-directory: libbzip2-rs-sys-cdylib
-        run: |
-          cargo build --release --target ${{matrix.target}} --no-default-features
-          cc -DNO_STD -o bzpipe bzpipe.c target/${{matrix.target}}/release/deps/libbz2_rs.so -I ../
-          ./bzpipe < Cargo.toml | ./bzpipe -d > out.txt
-          cmp -s Cargo.toml out.txt
```

Comment:
so now we never build wit h`-DNO_STD` right? we just always use the c allocator; in practice that means we test less (because the rust allocator was also dropped). 

Why make that change?

Reply:
this example is not really practical, it's just for demonstration and testing purposes.

Reply:
I can add back a `-DNO_STD` C example, but with this PR the cdylib will always be compiled with `#![no_std]`, which implies that we always have to use the C allocator as the Rust allocator is part of libstd. There is no benefit to building the cdylib with std. It only adds extra work and increases the size of the dylib.

Reply:
Added back testing of the no-stdio mode.